src/nvim/CMakeLists.txt

```diff
@@ -13,6 +13,9 @@ endif()
 if(WIN32)
   # tell MinGW compiler to enable wmain
   set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -municode")
+elseif(CMAKE_SYSTEM_NAME STREQUAL "Darwin")
+  set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -framework Foundation")
+  set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -framework Foundation")
```

Comment:
Sorry, my bad. It has to be `CoreFoundation`, not `Foundation`. Both exist, but the former is the low-level C API and the latter is a huge ObjC framework.